packages/testing-utils/src/index.js

```diff
@@ -18,20 +31,54 @@ export const cloneableGenerator = generatorFunc => (...args) => {
   }
 }
 
+const assertStatusRunning = status => {
+  if (status !== RUNNING) {
+    const str = statusToStringMap[status]
+    throw new Error(
+      `The task is no longer Running, it is ${str}. You can't change the status of a task once it is no longer running.`,
+    )
+  }
+}
+
 export function createMockTask() {
-  let _isRunning = true
-  let _result
-  let _error
+  let status = RUNNING
+  let taskResult
+  let taskError
 
   return {
     [TASK]: true,
-    isRunning: () => _isRunning,
-    result: () => _result,
-    error: () => _error,
-    cancel: () => {},
+    isRunning: () => status === RUNNING,
+    isCancelled: () => status === CANCELLED,
+    isAborted: () => status === ABORTED,
+    result: () => taskResult,
+    error: () => taskError,
+    cancel: () => {
+      assertStatusRunning(status)
+      status = CANCELLED
+    },
+    joiners: [],
```

Comment:
I have actually never used this mock task myself - it was here when I got involved in the project and I never bothered to tweak/remove it. I'm also very rusty when it comes to this codebase as I wasn't actively using it for a couple of last years. I wonder though - how useful this mock is in its current shape. 

It seems to me that this now won't throw when passed to a `join` but at the same time the `join` effect will never get resolved. 

I wonder if `setError` works OK when it comes to aborting the parent task etc. Do we have any tests for that? Does it work?

Reply:
All of the new tests in this PR are pretty comprehensive IMO of describing what behavior works vs what doesn't. If you set the error before it's passed to a join effect it will correctly throw and abort the parent task.